### ISSUE TITLE
Deleting a parent tag while some actor has the child tag

### ISSUE THREAD

**Issue by alcoranpaul** [FlaxEngine/FlaxEngine#2547]
**Issue description:**
When deleting a tag, in Layers and Tag setting, and some actor has that tag, the actor properties panel still have that tag existing.

**Steps to reproduce:**
1. Add a tag via Properties panel of some actor and **do not add** that tag to the actor
2. Add another tag but it will be the child of the previous tag and **add that tag** to the actor
3. Delete the parent tag via Layers and Tag Setting
4. Reload project (since deleting a tag in the Settings will not change in real time)
5. Go to the Actor that was added the tag and see that it is glitched 

- Layers and Tag setting would not have the parent tag but the child would still exists resulting in a glitched scenario
![image](https://github.com/FlaxEngine/FlaxEngine/assets/67833935/46c40dde-e3d4-4029-be7b-47cc1521e95e)

- Upon deleting the either the parent/child/both tag will remove the glitch.
> - but while **either or both tags is has not been deselected on an Actor**, it will result in Actor Properties panel to still show that deleted parent/child/both tag
![image](https://github.com/FlaxEngine/FlaxEngine/assets/67833935/52574aa3-44d1-47d9-8705-f11355dbd592)

- Also, in my project there are custom JSON assets that has a Tag parameter. 
> - it shows the same behavior as the Actor properties panel

**Workaround**
Deselect the desired Tag and Delete it from Tags and Layers setting

**Expected**
- When deleting a parent tag/or some tag in Layers and Tag setting while some actor has the child tag, will result in that tag being removed in real time and will be reflected the project. (Ex. In Actors properties panel and JSON assets tag ref)

**Proposed solution**
When removing a tag in Layers and Tag setting it will (In real time without reloading project)
>- Remove that child/parent tag that some actors has reference to
>- or A prompt that warns the developer that there are some actors that has the child tag and the parent cannot be deleted

**Flax version:**
Master

**Comment by alcoranpaul** [comment]
What I mean by editor property is this in Actor or JSON asset
![image](https://github.com/FlaxEngine/FlaxEngine/assets/67833935/6608181f-b461-4ef7-a23f-3eed7c5d818a)

**Comment by mafiesto4** [comment]
The current design allows assets from game and plugin projects to provide tags without need to specify them in layers&tags asset. Maybe here we could scan content and remove tags from content when deleting them from the layers&tags asset.

**Comment by ptcsoderlund** [comment]
Seems like this bug is still here.
It looks like a memory cache problem.
I add tags and they show up.
If i remove tags from settings file they are not removed from inspector tags. Which i believe is stored in memory?
When i restart editor, the tags from settings is getting setup correctly in inspector.
But changes to tags in inspector dissapears when reloading editor.
So I have to make sure to restart editor before setting tags on actors when i have removed or changed tags in inspector/tagfile.

**Comment by mafiesto4** [comment]
Yeah, it's a known thing about the editor restart. Maybe we could solve it somehow.
